### PR TITLE
Feature/icon requirement updates

### PR DESCRIPTION
**Pull request by jennyvb143**
Updates the `Icon` component per the following new requirements:
Require role prop
Set default role to decorative
Require ariaLabel when role is set to img
Update all Components where `Icon` is a dependency to add `role` prop and `ariaLabel` prop as needed
Add all icons currently in use on ebth-com repo

Clubhouse Story: https://app.clubhouse.io/ebth/story/2019/update-icon-component

### CODE REVIEW

**Comment by tonytrucco-CS** [comment]
Hmm, now all checks fail? What can be done about these?

**Review by tonytrucco-CS — APPROVED** [review]
Looks good. Just the weird failing tests.

**Comment by jennyvb143** [comment]
> Looks good. Just the weird failing tests.

Oops. Forgot to update the tests. Update forthcoming!

**Comment by jennyvb143** [comment]
These tests were failing because the newly required `role` prop was missing from all `Components` where `Icon` is a dependency. In most cases, `Icon` is used inside `Tooltip`. In these instances, I could only get the screen reader to read/access the tooltip content if I used `role="img"` (even when inside a focusable `input` tag) and it still seems to be spotty in terms of reading the content every time. So I think we need to have a larger discussion re: accessibility for the tooltip at some point. @tonytrucco-CS I used `Tooltip Icon` for the required `ariaLabel` prop, so please let me know if you think that needs to be changed.

In all other instances (`Icon` NOT inside `Tooltip`), I made a judgement call to assign `decorative` to the `role` prop. It seemed to be the best choice but @tonytrucco-CS please review for accuracy! 😊

**Review by tonytrucco-CS — APPROVED** [review]
I think it looks good. Hard to say completely until we get it into the field, but I think you've covered the bases pretty good here. Tooltips are notoriously finicky.

File: src/atoms/Icon/Icon.tsx [hunk 1]
```diff
@@ -5,15 +5,20 @@ import { StyledIcon } from './Icon.style';
...
-export type IconProps = Props &
+interface DecorativeProps extends Props {
+  role: 'decorative';
+  ariaLabel?: string;
+}
+interface ImgProps extends Props {
+  role: 'img';
+  ariaLabel: string;
+}
+export type IconProps = (DecorativeProps | ImgProps) &
```
Comment (mosherc) [inline src/atoms/Icon/Icon.tsx:21]: Nit: blank lines between interfaces and types

File: src/atoms/Icon/Icon.stories.tsx [hunk 2]
```diff
@@ -5,11 +5,10 @@ import { Icon, IconProps } from './Icon';
 import { icons } from './icons';
 
 type StoryType = {
-  ariaLabel?: string;
+  role: 'decorative';
```
Comment (mosherc) [inline src/atoms/Icon/Icon.stories.tsx:8]: I think this is supposed to be: 
```suggestion
  role: 'decorative' | 'img';
```

File: src/atoms/Icon/Icon.tsx [hunk 3]
```diff
@@ -5,15 +5,20 @@ import { StyledIcon } from './Icon.style';
...
   color?: string;
 }
-
-export type IconProps = Props &
+interface DecorativeProps extends Props {
+  role: 'decorative';
+  ariaLabel?: string;
+}
+interface ImgProps extends Props {
+  role: 'img';
```
Comment (mosherc) [inline src/atoms/Icon/Icon.tsx:18]: I wonder if we can use JSDocs to document the `role` prop so consumers can see when they should choose decorative or img based on a description written here? I've been messing around on this branch and can't quite figure out how to document _just_ the role prop and not the whole interface or exported component...we can look into this later.

**Review by mosherc — APPROVED** [review]

**Pull request merged**